modules/hivemq/src/main/java/org/testcontainers/hivemq/HiveMQContainer.java

```diff
@@ -0,0 +1,651 @@
+                        hiveMQExtension.getStartPriority()),
+                Charset.defaultCharset());
+
+        if (hiveMQExtension.isDisabledOnStartup()) {
+            final File disabled = new File(extensionDir, "DISABLED");
+            final boolean newFile = disabled.createNewFile();
+            if (!newFile) {
+                throw new ContainerLaunchException("Could not create DISABLED file '" + disabled.getAbsolutePath() + "' on host machine.");
+            }
+        }
+
+        final JavaArchive javaArchive =
+                ShrinkWrap.create(JavaArchive.class)
+                    .addAsServiceProvider(EXTENSION_MAIN_CLASS_NAME, hiveMQExtension.getMainClass().getName());
+
+        putSubclassesIntoJar(hiveMQExtension.getId(), hiveMQExtension.getMainClass(), javaArchive);
+        for (final Class<?> additionalClass : hiveMQExtension.getAdditionalClasses()) {
+            javaArchive.addClass(additionalClass);
+            putSubclassesIntoJar(hiveMQExtension.getId(), additionalClass, javaArchive);
+        }
+
+        javaArchive.as(ZipExporter.class).exportTo(new File(extensionDir, "extension.jar"));
+
+        return extensionDir;
+    }
+
+    private void putSubclassesIntoJar(
+            final @NotNull String extensionId,
+            final @Nullable Class<?> clazz,
+            final @NotNull JavaArchive javaArchive) throws NotFoundException {
+
+        if (clazz != null) {
+            final Set<String> subClassNames =
+                    ClassPool.getDefault().get(clazz.getName()).getClassFile().getConstPool().getClassNames();
+            for (final String subClassName : subClassNames) {
+                final String className = subClassName.replaceAll("/", ".");
+
+                if (!className.startsWith("[L")) {
+                    logger.debug("Trying to package subclass '{}' into extension '{}'.", className, extensionId);
+                    javaArchive.addClass(className);
+                } else {
+                    logger.debug("Class '{}' will be ignored.", className);
+                }
+            }
+        }
+    }
+
+    /**
+     * Removes the specified prepackaged extension folders from '/opt/hivemq/extensions' before the container is built.
+     * Note: this creates a custom docker image.
+     * <p>
+     * Must be called before the container is started.
+     *
+     * @param extensionIds the prepackaged extensions to remove
+     * @return self
+     */
+    public @NotNull HiveMQContainer withoutPrepackagedExtensions(final @NotNull String... extensionIds) {
+        final String dockerImageName = getDockerImageName();
+        setImage(new ImageFromDockerfile(dockerImageName + "-custom")
+                .withDockerfileFromBuilder(builder -> {
+                    builder.from(dockerImageName);
+                    for (final String extensionId : extensionIds) {
+                        builder.run("rm", "-rf", "/opt/hivemq/extensions/" + extensionId);
```

Comment:
I wonder if instead of building a custom Docker image, the UX would be better if we override the startup command and remove them before application startup?

Reply:
Unfortunately this does not work. 

If you add a HiveMQ Extension with `withExtension(...)` and add `rm -rf /opt/hivemq/extensions/` to the startup command the added HiveMQ Extension will also be removed. :(

Reply:
OK let's see how this goes - I do think that this might cause problems in edge cases, but maybe we can see in practice whether people actually encounter it. At least the API should presumably not change if we decide to change how it's done.